cmake/schemes/url_sha1_boost_library.cmake.in

```diff
@@ -102,6 +102,23 @@ else()
   hunter_fatal_error("TODO: set toolset for boost" WIKI "error.boost.toolset")
 endif()
 
+if("@ANDROID@")
+  set(toolset_version ndk)
+endif()
+
+get_directory_property(defs COMPILE_DEFINITIONS)
+foreach(def ${defs})
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D${def}")
```

Comment:
Well we know that it's Android toolchain specific behaviour. Both variants are good for me. If you want to make it independent of the toolchain then you can introduce new variable (may be there is already some CMake variable?), say `DEFINITION_OPTION`:
```cmake
if(MSVC)
  set(DEFINITION_OPTION "/D")
else()
  set(DEFINITION_OPTION "-D")
endif()

set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${DEFINITION_OPTION}${def}")
```